tests/structured_conf/test_structured_config.py

```diff
@@ -1051,3 +1051,216 @@ def test_update_userlist(
+            ("ContainerInDict", {"users": [1, True]}, "dict_with_list"),
+            ("ContainerInDict", {"users": [1, User()]}, "dict_with_list"),
+            ("ComplexList", [[{"number": True}]], "list"),
+            ("ComplexList", [[{"number": User()}]], "list"),
+            ("ComplexDict", {"users": {"data": True}}, "dict"),
+            ("ComplexDict", {"users": {"data": User()}}, "dict"),
+        ],
+    )
+    def test_container_as_element_type_invalid(
+        self, class_type: str, obj: str, value: Any, node: str
+    ) -> None:
+        module: Any = import_module(class_type)
+        class_ = getattr(module, obj)
+        cfg = OmegaConf.structured(class_)
+        with pytest.raises(ValidationError):
+            cfg[node] = value
+
+    @pytest.mark.parametrize(  # type: ignore
+        "obj,cfg2,expected",
+        [
+            (
+                "ContainerInList",
+                OmegaConf.create(
+                    {"list_with_list": [[4, 5]], "list_with_dict": [{"foo": 4}]}
+                ),
+                {
+                    "list_with_list": [[4, 5]],
+                    "list_with_dict": [{"foo": 4}],
+                },
+            ),
+            (
+                "ContainerInDict",
+                OmegaConf.create(
+                    {
+                        "dict_with_list": {"foo": [3, 4]},
+                        "dict_with_dict": {"foo": {"var": 4}},
+                    }
+                ),
+                {
+                    "dict_with_list": {"foo": [3, 4]},
+                    "dict_with_dict": {"foo": {"var": 4}},
+                },
+            ),
+            (
+                "ComplexList",
+                OmegaConf.create(
+                    {
+                        "list": [[{"foo": 13}]],
+                    }
+                ),
+                {
+                    "list": [[{"foo": 13}]],
+                },
+            ),
+            (
+                "ComplexDict",
+                OmegaConf.create(
+                    {
+                        "dict": {"foo2": {"var2": [3, 4]}},
+                    }
+                ),
+                {
+                    "dict": {"foo": {"var": [1, 2]}, "foo2": {"var2": [3, 4]}},
+                },
+            ),
+        ],
+    )
+    def test_container_as_element_type_merge(
+        self, class_type: str, obj: str, cfg2: Any, expected: Any
+    ) -> None:
+        module: Any = import_module(class_type)
+        class_ = getattr(module, obj)
+        cfg = OmegaConf.merge(class_)
+        res = OmegaConf.merge(cfg, cfg2)
+        assert res == expected
+
+    @pytest.mark.parametrize(  # type: ignore
+        "obj,cfg2",
+        [
+            (
+                "ContainerInList",
+                OmegaConf.create(
+                    {
+                        "list_with_list": [[4, "invalid_value"]],
+                        "list_with_dict": [{"foo": 4}],
+                    }
+                ),
```

Comment:
Instead of passing and DictConfig pass the dict itself and wrap it in the test function.